Branch-SDK/src/io/branch/indexing/ContentDiscoverer.java

```diff
@@ -151,83 +169,185 @@ public void run() {
+        JSONObject listViewContentObj = new JSONObject();
+        contentKeysArray.put(viewKeyString);
+        contentDataArray.put(listViewContentObj);
+        viewKeyString = viewKeyString.replace(COLLECTION_VIEW_KEY_PREFIX, "");
+        try {
+            JSONObject viewKeyJson = new JSONObject(viewKeyString);
+            if (viewKeyJson.length() > 0) {
+                String listViewID = viewKeyJson.keys().next();
+                int id = activity.getResources().getIdentifier(listViewID, "id", activity.getPackageName());
+                // PRS : in case of View pager or Tabbed controls there may be many tabs. The same control can be included in more than one tab
+                // Need to make sure the focused content is checked in case it is available or fallback to the root view
+                View view = activity.getCurrentFocus() != null ? activity.getCurrentFocus().findViewById(id) : null;
+                if (view == null) {
+                    view = activity.findViewById(id);
+                }
+                if (view != null && view instanceof ViewGroup) {
+                    ViewGroup listView = (ViewGroup) view;
+                    JSONArray itemViewChildIdArray = viewKeyJson.getJSONArray(listViewID);
+                    int itemViewIds[] = new int[itemViewChildIdArray.length()];
+                    for (int i = 0; i < itemViewChildIdArray.length(); i++) {
+                        itemViewIds[i] = activity.getResources().getIdentifier(itemViewChildIdArray.getString(i), "id", activity.getPackageName());
+                    }
+                    int firstVisibleItemIdx = listView instanceof AbsListView ? ((AbsListView) listView).getFirstVisiblePosition() : 0;
+                    for (int j = 0; j < listView.getChildCount(); j++) {
+                        if (listView.getChildAt(j) != null) {
+                            JSONObject itemObj = new JSONObject();
+                            listViewContentObj.put("" + (j + firstVisibleItemIdx), itemObj);
+                            for (int i = 0; i < itemViewIds.length; i++) {
+                                if (listView.getChildAt(j) != null) {
+                                    View itemViewChild = listView.getChildAt(j).findViewById(itemViewIds[i]);
+                                    if (itemViewChild instanceof TextView) {
+                                        itemObj.put(itemViewChildIdArray.getString(i), getTextViewValue(itemViewChild, isClearText));
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    boolean isScrollListeningEnabled = viewKeyJson.has(ENABLE_SCROLL_WATCH) && viewKeyJson.getBoolean(ENABLE_SCROLL_WATCH);
+                    if (isScrollListeningEnabled && !viewTreeObserverMap.containsKey(viewKeyString)) {
+                        listView.getViewTreeObserver().addOnScrollChangedListener(scrollChangedListener);
```

Comment:
since we're doing this in a "loop", are we going to add a lot of scroll change listeners to their view?